pkg/virt-controller/watch/vm.go

```diff
@@ -586,22 +581,14 @@ func (c *VMController) generateVMIMemoryDumpVolumePatch(vmi *virtv1.VirtualMachi
 	} else {
 		vmiCopy.Spec = *removeMemoryDumpVolumeFromVMISpec(&vmiCopy.Spec, request.ClaimName)
 	}
-
-	oldJson, err := json.Marshal(vmi.Spec.Volumes)
-	if err != nil {
-		return err
-	}
-
-	newJson, err := json.Marshal(vmiCopy.Spec.Volumes)
+	patch, err := patch.New(
+		patch.WithTest("/spec/volumes", vmi.Spec.Volumes),
+		patch.WithReplace("/spec/volumes", vmiCopy.Spec.Volumes),
+	).GeneratePayload()
```

Comment:
Hi, first time looking at the `patch` addition, this is neat.

Previously, if `vmi.Spec.Volumes` was 0 the operation would be `add` instead of `replace` but it feels that `replace` is fine for both scenarios. I wonder why before was `add`.

Reply:
I have thought a little bit more about it and it might not be safe.
If the key volumes is always present then the patch with `replace` will always work, but if volumes is null the replace is going to be ignored.
I'll restore the Add/Replace logic